src/net_processing.cpp

```diff
@@ -2572,33 +2579,79 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 recentRejects->insert(tx.GetHash());
             }
         } else {
-            assert(IsTransactionReason(state.GetReason()));
-            if (!tx.HasWitness() && state.GetReason() != ValidationInvalidReason::TX_WITNESS_MUTATED) {
-                // Do not use rejection cache for witness transactions or
-                // witness-stripped transactions, as they can have been malleated.
-                // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
-                assert(recentRejects);
-                recentRejects->insert(tx.GetHash());
-                if (RecursiveDynamicUsage(*ptx) < 100000) {
-                    AddToCompactExtraTransactions(ptx);
+            // If this tx didn't make it in due to feerate, and there is a tx
+            // in the orphan pool -- then maybe that tx is only missing this
+            // one parent.
+            // Try to process the pair as a package.
+            bool added_as_package = false;
+            if (state.GetRejectCode() == REJECT_INSUFFICIENTFEE) {
```

Comment:
Maybe also add a check for `g_orphan_list.size() > 0` here? But I am not experienced with the orphan pool so it may be fair to leave it out if the orphan pool is populated 99% of the time.